### PR TITLE
Fix missing PTU by exchg_after_x_days trigger

### PR DESCRIPTION
**Pull request by tkireva**
Wrong fix to old defect related to redundant PTU is removed.
As a fix first pt_exchanged_x_days_after_epoch is checked if
it is initialized.

### CODE REVIEW

**Comment by tkireva** [comment]
Please review:
@okozlovlux @istoilovgithub @LuxoftAKutsan @LitvinenkoIra

File: src/components/policy/policy_regular/src/cache_manager.cc [hunk 1]
```diff
@@ -564,16 +564,22 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
 int CacheManager::DaysBeforeExchange(int current) {
   LOG4CXX_AUTO_TRACE(logger_);
   CACHE_MANAGER_CHECK(0);
+
+  const rpc::Optional<rpc::Integer<uint16_t, 0, 65535> > days_after_epoch =
+      (pt_->policy_table.module_meta->pt_exchanged_x_days_after_epoch);
+
```
Comment (indlu) [inline src/components/policy/policy_regular/src/cache_manager.cc]: @tkireva, you could use Optional from "utils/optional.h" ,
something like optional::Optional<uint32_t> days_after_epoch

File: src/components/policy/policy_external/src/cache_manager.cc [hunk 2]
```diff
@@ -1221,16 +1221,22 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
 int CacheManager::DaysBeforeExchange(uint16_t current) {
```
Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/cache_manager.cc]: @tkireva if you use enum in return. Please use it in sugnature

File: src/components/policy/policy_regular/src/cache_manager.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/policy/policy_regular/src/cache_manager.cc]: @indlu , @tkireva should use the same type as in PT. So it is ok that is is Optional from POlicies.

File: src/components/policy/policy_external/src/cache_manager.cc (hunk 2 above)
Comment (tkireva) [inline src/components/policy/policy_external/src/cache_manager.cc]: @LuxoftAKutsan Done in 10ed3ebecd48fbe9dccfb60520d2ebeef411b54b

**Comment by tkireva** [comment]
Rebuild required

**Comment by tkireva** [comment]
Rebuild required

**Pull request merged**